ggml/src/ggml-cuda/common.cuh

```diff
@@ -964,6 +965,38 @@ struct ggml_cuda_graph {
 #endif
 };
 
+struct ggml_cuda_concurrent_event {
+    std::vector<cudaEvent_t> join_events;
+    cudaEvent_t              fork_event;
+
+    int                                          n_streams = 0;
+    std::unordered_map<const ggml_tensor *, int> stream_mapping;
+
+    const ggml_tensor * join_node;
+
+    ggml_cuda_concurrent_event() = default;
+
+    explicit ggml_cuda_concurrent_event(int n_streams) : n_streams(n_streams) {
+        join_events.resize(n_streams);
+
+        for (size_t i = 0; i < join_events.size(); ++i) {
+            CUDA_CHECK(cudaEventCreateWithFlags(&join_events[i], cudaEventDisableTiming));
+        }
+
+        CUDA_CHECK(cudaEventCreateWithFlags(&fork_event, cudaEventDisableTiming));
+    }
+};
+
+struct ggml_cuda_stream_context {
+    std::vector<const ggml_tensor *> original_graph;
```

Comment:
```suggestion
    std::vector<const ggml_tensor *> original_nodes;
```

A "graph" in the context of ggml consists of both "node" and "leaf" tensors so I would suggests this name to be more clear.